src/minisketch.cpp

```diff
@@ -130,12 +132,49 @@ uint32_t minisketch_implementation_max() {
 }
 
 minisketch* minisketch_create(uint32_t bits, uint32_t implementation, size_t capacity) {
-    Sketch* sketch = Construct(bits, implementation);
-    if (sketch) {
-        sketch->Init(capacity);
-        sketch->Ready();
+    try {
```

Comment:
I think this can be written more readably as

```c++
Sketch* sketch = nullptr;
try {
    sketch = Construct(bits, implementation);
    sketch->Init(capacity);
    sketch->Ready();
    return (minisketch*)sketch;
} catch (std::bad_alloc&) {
    if (sketch) delete sketch;
    return nullptr;
}
```